project-base/tests/ShopBundle/Functional/Model/Product/ProductVisibilityRepositoryTest.php

```diff
@@ -65,36 +65,50 @@ private function getDefaultProductData()
         /** @var \Shopsys\ShopBundle\Model\Category\Category $category */
         $category = $this->getReference(CategoryDataFixture::CATEGORY_ELECTRONICS);
 
-        $em = $this->getEntityManager();
-        $vatData = new VatData();
-        $vatData->name = 'vat';
-        $vatData->percent = '21';
-        $vat = new Vat($vatData);
-        $em->persist($vat);
-
         /** @var \Shopsys\ShopBundle\Model\Product\ProductData $productData */
         $productData = $this->productDataFactory->create();
         $names = [];
         foreach ($this->localization->getLocalesOfAllDomains() as $locale) {
             $names[$locale] = 'Name';
         }
         $productData->name = $names;
-        $productData->vat = $vat;
         $productData->categoriesByDomainId = [Domain::FIRST_DOMAIN_ID => [$category]];
         $productData->availability = $this->getReference(AvailabilityDataFixture::AVAILABILITY_IN_STOCK);
         $productData->unit = $this->getReference(UnitDataFixture::UNIT_PIECES);
         $this->setPriceForAllDomains($productData, Money::create(100));
+        $this->setVatsForAllDomains($productData);
 
         return $productData;
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductData $productData
+     */
+    private function setVatsForAllDomains(ProductData $productData): void
+    {
+        $productVats = [];
+        foreach ($this->domain->getAllIds() as $domainId) {
+            $em = $this->getEntityManager();
```

Comment:
If you want to save entityManager to variable, please do it outside of loop

Reply:
Fixed